solr/core/src/java/org/apache/solr/cloud/gumi/SolrCollection.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.cloud.gumi;
+
+import java.util.Set;
+
+/**
+ * Represents a Collection in SolrCloud. Although naming this class "Collection" is possible it would be confusing.
+ */
+public interface SolrCollection {
```

Comment:
Why don't we use the standard DocCollection class ?

Reply:
Same as above for `Replica`, here the class is even more complex.
For example we've been having discussions on getting rid of the term "Slice" and standardizing on "Shard". If we expose `DocCollection` in its current state, `Slice` can't be renamed.

Also, if we want to build a simulation framework later for plugin developers, that layer will have a clean and easy interface to build on, since that's all the plugins are seeing. Building it will not require reaching into Solr code in various places and add hooks for the simulator. And removing it will be a lot easier as well :)

Reply:
BTW the two sample plugins ([SamplePluginMinimizeCores](https://github.com/apache/lucene-solr/pull/1684/files#diff-ddbe185b5e7922b91b90dfabfc50df4c) and [SamplePluginRandomPlacement](https://github.com/apache/lucene-solr/pull/1684/files#diff-a707d47e9a1c89551d455e2ad167f5a5)) show how the interfaces are used.

If we wanted to use existing classes instead of some of the interfaces in the proposal (even though I think it's a bad idea because it will make refactoring challenging), we can easily play with the various interfaces and see what it means and how the plugin code evolves.